netstandard/tools/HandlePackageFileConflicts.netstandard.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public partial class HandlePackageFileConflicts : Task
+    {
+        private Version GetAssemblyVersion(string sourcePath)
+        {
+            using (var assemblyStream = new FileStream(sourcePath, FileMode.Open, FileAccess.Read, FileShare.Delete | FileShare.Read))
+            {
+                Version result = null;
+                try
+                {
+                    using (PEReader peReader = new PEReader(assemblyStream, PEStreamOptions.LeaveOpen))
+                    {
+                        if (peReader.HasMetadata)
+                        {
+                            MetadataReader reader = peReader.GetMetadataReader();
+                            if (reader.IsAssembly)
+                            {
+                                result = reader.GetAssemblyDefinition().Version;
+                            }
+                        }
+                    }
+                }
+                catch (BadImageFormatException)
```

Comment:
What throws this error? We're not loading the assembly, just reading the file and you already check HasMetadata